src/WeightList.php

```diff
@@ -25,6 +25,7 @@ public function set($value, $weight)
     private function getValueByPosition($position)
     {
         $accumulator = 0;
+        $value = null;
```

Comment:
looks like my bug. if list is empty, and you try to get in's value by position, it will be incorrect to silently return null, because null may be value. this is gook case for `OutOfBoundsException` http://php.net/manual/ru/class.outofboundsexception.php

Reply:
It's good to replace ```return value``` with ```OutOfBoundException``` on the last line of the method.